common/cross_sockets/XSocketServer.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "common/cross_sockets/XSocket.h"
+
+#include <thread>
+#include "common/common_types.h"
+#include <functional>
+#include <mutex>
+
+/// @brief A cross platform generic socket server implementation
+class XSocketServer {
+ public:
+  static constexpr int DEF_BUFFER_SIZE = 32 * 1024 * 1024;
+  XSocketServer(std::function<bool()> shutdown_callback,
+                int _tcp_port,
+                int _buffer_size = DEF_BUFFER_SIZE);
+  ~XSocketServer();
```

Comment:
`virtual ~XSocketServer();`
it's a good idea to make the destructor virtual if you are using virtual methods.